### PR TITLE
Resource quota bug fixes

### PR DESCRIPTION
**Pull request by alena1108**
https://github.com/rancher/rancher/issues/15336
https://github.com/rancher/rancher/issues/15338
https://github.com/rancher/rancher/issues/15318
https://github.com/rancher/rancher/issues/14835

### CODE REVIEW

File: pkg/api/customization/project/project_store.go [hunk 1]
```diff
@@ -31,11 +39,26 @@ func (s *projectStore) Create(apiContext *types.APIContext, schema *types.Schema
...
 	values.PutValue(data, annotation, "annotations", roleTemplatesRequired)
 
 	return s.Store.Create(apiContext, schema, data)
 }
 
+func (s *projectStore) Update(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}, id string) (map[string]interface{}, error) {
+	if err := s.validateResourceQuota(apiContext, schema, data, id); err != nil {
+		return nil, err
+	}
+	if err := s.validateResourceQuota(apiContext, schema, data, ""); err != nil {
```
Comment (cjellick) [inline pkg/api/customization/project/project_store.go]: WHy do you call this twice back-to-back with just the id paramter differing?
Seems like the implementation of validateResourceQuota doesnt use the id parameter at all.

File: pkg/api/customization/project/project_store.go [hunk 2]
```diff
@@ -3,14 +3,22 @@ package project
 import (
 	"encoding/json"
```
Comment (cjellick) [inline pkg/api/customization/project/project_store.go]: nit: Unnecessary newlines between std lib packages

File: pkg/api/store/namespace/store.go [hunk 3]
```diff
@@ -36,5 +48,86 @@ func (p *Store) Create(apiContext *types.APIContext, schema *types.Schema, data
...
+
+func (p *Store) Update(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}, id string) (map[string]interface{}, error) {
+	if err := p.validateResourceQuota(apiContext, schema, data, id); err != nil {
+		return nil, err
+	}
+
+	return p.Store.Update(apiContext, schema, data, id)
+}
+
+func (p *Store) validateResourceQuota(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}, id string) error {
```
Comment (cjellick) [inline pkg/api/store/namespace/store.go]: id field not used again?

File: pkg/api/customization/project/project_store.go (hunk 1 above)
Comment (alena1108) [reply pkg/api/customization/project/project_store.go]: @cjellick copy/paste error :) will fix

**Comment by cjellick** [comment]
LGTM

**Pull request merged**